go/adbc/driver/databricks/schema.go

```diff
@@ -128,182 +194,287 @@ func ResultSchemaToArrowSchema(dbxSchema *sql.ResultSchema) (*arrow.Schema, erro
+		keyType, valueType, err := parseMapKeyValue(content)
+		if err != nil {
+			return nil, fmt.Errorf("invalid map type format: %w", err)
+		}
+		keyArrowType, err := getArrowTypeFromStringType(keyType)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse map key type: %w", err)
+		}
+		valueArrowType, err := getArrowTypeFromStringType(valueType)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse map value type: %w", err)
+		}
+		return arrow.MapOf(keyArrowType, valueArrowType), nil
+	}
+	// INTERVAL types - comprehensive pattern for all interval types
+	if matches := regexp.MustCompile(intervalTypeRegex).FindStringSubmatch(col); matches != nil {
+		// Year-month intervals
+		if strings.Contains(matches[1], "YEAR") || matches[1] == "MONTH" {
+			return arrow.FixedWidthTypes.MonthInterval, nil
+		}
+		// All other intervals are day-time intervals
+		return arrow.FixedWidthTypes.Duration_s, nil
+	}
+	// STRUCT < [fieldName [:] fieldType [NOT NULL] [COLLATE collationName] [COMMENT str] [, …] ] >
+	if content, ok := extractOutermostBracketContent(col, "STRUCT"); ok {
+		fields, err := parseStructFields(content)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse struct: %w", err)
+		}
+		arrowFields := make([]arrow.Field, 0, len(fields))
+		for _, field := range fields {
+			fieldName, fieldType, nullable, err := parseStructField(field)
 			if err != nil {
-				return nil, fmt.Errorf("failed to parse map key type: %w", err)
+				return nil, fmt.Errorf("failed to parse struct field '%s': %w", field, err)
 			}
-			valueArrowType, err := getArrowTypeFromColumnInfo(valueCol)
+			fieldArrowType, err := getArrowTypeFromStringType(fieldType)
 			if err != nil {
-				return nil, fmt.Errorf("failed to parse map value type: %w", err)
+				return nil, fmt.Errorf("failed to parse struct field type %s: %w", fieldName, err)
 			}
-
-			return arrow.MapOf(keyArrowType, valueArrowType), nil
+			arrowFields = append(arrowFields, arrow.Field{
+				Name:     fieldName,
+				Type:     fieldArrowType,
+				Nullable: nullable,
+			})
 		}
-		return nil, fmt.Errorf("invalid map type format: %v", col.TypeName)
-	case sql.ColumnInfoTypeNameStruct:
-		// todo(jasonlin45): Full support for struct field. No support for NOT NULL, COLLATE or COMMENT available
-		// STRUCT < [fieldName [:] fieldType [NOT NULL] [COLLATE collationName] [COMMENT str] [, …] ] >
-		if matches := regexp.MustCompile(`STRUCT<(.+)>`).FindStringSubmatch(col.TypeText); matches != nil {
-			fieldsStr := matches[1]
-
-			fields := []string{}
-
-			// fields are comma separated, nested types are in parentheses
-			fieldRegex := regexp.MustCompile(`([^,()]+(?:\([^()]*\)[^,()]*)*)`)
-			matches := fieldRegex.FindAllString(fieldsStr, -1)
-			for _, field := range matches {
-				fields = append(fields, strings.TrimSpace(field))
-			}
+		return arrow.StructOf(arrowFields...), nil
+	}
+	// default to an unrecognized type so we can still positionally propogate metadata
```

Comment:
```suggestion
    // Use Null to occupy schema column position, so metadata remains correctly aligned even when the type is unrecognized.
```

maybe this? words har.